plutus-chain-index/app/Marconi/Logging.hs

```diff
@@ -0,0 +1,120 @@
+      modifyIORef' statsRef $ \stats ->
+        stats {syncStatsNumBlocks = syncStatsNumBlocks stats + 1}
+      printMessage statsRef cp ct
+    update statsRef (RollBackward cp ct) = do
+      modifyIORef' statsRef $ \stats ->
+        stats {syncStatsNumRollbacks = syncStatsNumRollbacks stats + 1}
+      printMessage statsRef cp ct
+
+    printMessage statsRef cp ct = do
+      SyncStats {syncStatsNumBlocks, syncStatsNumRollbacks, syncStatsLastMessage} <- readIORef statsRef
+
+      now <- getCurrentTime
+
+      let timeSinceLastMsg = diffUTCTime now <$> syncStatsLastMessage
+
+      let blocksMsg = case timeSinceLastMsg of
+            Nothing -> id
+            Just t -> \k ->
+              "Processed"
+                <+> pretty syncStatsNumBlocks
+                <+> "blocks in the last"
+                <+> pretty (formatTime defaultTimeLocale "%s" t)
+                <+> "seconds"
+                <+> let rate = fromIntegral syncStatsNumBlocks / realToFrac t :: Double
+                     in pretty (printf "(%.0f blocks/sec)." rate :: String)
+                <+> k
+
+      let rollbackMsg = case timeSinceLastMsg of
+            Nothing -> id
+            Just t -> \k ->
+              ( case syncStatsNumRollbacks of
+                  0 -> "No"
+                  _ -> pretty syncStatsNumRollbacks
+              )
+                <+> "rollbacks in the last"
+                <+> pretty (formatTime defaultTimeLocale "%s" t)
+                <+> "seconds."
+                <+> k
+
+      let syncMsg = case (cp, ct) of
+            (ChainPoint (SlotNo chainPointSlot) _, ChainTip (SlotNo chainTipSlot) _header _blockNo)
+              -- TODO: MAGIC number here. Is there a better number?
+              -- 100 represents the number of slots before the
+              -- node where we consider the chain-index to be synced.
+              | chainTipSlot - chainPointSlot < 100 ->
+                "Synchronised."
+            (ChainPoint (SlotNo chainPointSlotNo) _, ChainTip (SlotNo chainTipSlotNo) _header _blockNo) ->
+              let pct = (100 :: Double) * fromIntegral chainPointSlotNo / fromIntegral chainTipSlotNo
+               in pretty
+                    ( printf
+                        "Synchronising. Current slot %d out of %d (%0.2f%%)."
+                        chainPointSlotNo
+                        chainTipSlotNo
+                        pct ::
+                        String
+                    )
+            _ -> "Starting."
+
+      let shouldPrint = case timeSinceLastMsg of
+            Nothing -> True
+            Just t
+              | t > minSecondsBetweenMsg -> True
+              | otherwise -> False
+
+      when shouldPrint $ do
+        logInfo tracer $
+          renderStrict $
+            layoutPretty defaultLayoutOptions $
+              syncMsg <+> (blocksMsg $ rollbackMsg $ "Chain tip is" <+> pretty ct <> ".")
```

Comment:
We want to know the chain tip of the synced block, not the tip of the node. So it should be `pretty cp`.